### PR TITLE
Test option sets

### PR DESCRIPTION
**Pull request by noryb009**
This adds some basic option set tests. It ensures methods can be
added to option sets using limit files, command line arguments
can be given to option sets, and those options are correctly applied.

Option sets are a way to apply certain command line arguments to
groups of specific methods being compiled. Each method can optionally
be placed into one of these sets by annotating a limit file.

Methods can be added by modifying their successful compilation message
(starting with '+') in a limit file.

    + (warm @ ...) File.cpp:123:myMethod

is not in any option set. It can be added to set 1 by adding a '1'
directly after the '+':

    +1 (warm @ ...) File.cpp:123:myMethod

There are 10 option sets methods can be placed into, numbered 0 to 9.
By default, methods are in option set 0.

Command line arguments can be targeted towards a certain option set by
placing the argument in brackets after the set number. For example,

    -Xjit:optlevel=cold,2(optlevel=warm),7(optlevel=hot)

will compile all methods in option set 2 as warm, all methods in set 7
as hot, and all other methods as cold.

### CODE REVIEW

**Comment by vijaysun-omr** [comment]
I would like @mpirvu to review this.

File: fvtest/compilertest/tests/LimitFileTest.cpp [hunk 1]
```diff
@@ -96,13 +102,14 @@ TestCompiler::LimitFileTest::generateVLog(const char *vlog, const char *limitFil
  *
  * @param[in] vlog The log from the test.
  * @param[in] method The name of the method to search for.
- * @param[in] compiled Assert that the method was or was not compiled.
- * @param[out] The line of the log that the method is on. This can be
- *             used in a limit file parameter. This parameter is only
- *             valid when compiled is true.
+ * @param[in] compiled Assert the level the method was compiled at.
```
Comment (mpirvu) [inline fvtest/compilertest/tests/LimitFileTest.cpp]: 'compiled' --> 'level'

File: fvtest/compilertest/tests/LimitFileTest.cpp [hunk 2]
```diff
@@ -96,13 +102,14 @@ TestCompiler::LimitFileTest::generateVLog(const char *vlog, const char *limitFil
...
+ *            Using NULL asserts the method was not compiled.
+ * @param[out] foundOnLine The line of the log that the method is on.
+ *             This can be used in a limit file parameter. This
+ *             parameter is only valid when compiled is true.
  */
 void
-TestCompiler::LimitFileTest::checkVLogForMethod(const char *vlog, const char *method, bool compiled, int *foundOnLine)
+TestCompiler::LimitFileTest::checkVLogForMethod(const char *vlog, const char *method, char *level, int *foundOnLine)
```
Comment (mpirvu) [inline fvtest/compilertest/tests/LimitFileTest.cpp]: 'char *level' should probably by a 'const char *level'

File: fvtest/compilertest/tests/OptionSetTest.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+   std::ofstream out(optSetFile);
+   ASSERT_TRUE(out.is_open());
+   delayUnlink(optSetFile);
+
+   std::string line;
+   while(std::getline(in, line))
+      {
+      // Limitfiles have a leading newline, but no newline on the end
+      out << '\n';
```
Comment (mpirvu) [inline fvtest/compilertest/tests/OptionSetTest.cpp]: Do we want to preserve empty lines?

**Comment by noryb009** [comment]
Each project can define a specific vlog line format - by default, OMR uses the format I mentioned in the commit message. In this format, `File.cpp:123:myMethod` is the method signature (again, the default format for OMR).

**Comment by noryb009** [comment]
I've addressed @mpirvu 's comments.

**Comment by mpirvu** [comment]
I am ok with the changes.

File: compiler/control/OMROptions.cpp [hunk 4]
```diff
@@ -3277,7 +3277,7 @@ OMR::Options::processOptionSet(
             endOpt = options + strlen(EXCLUDED_METHOD_OPTIONS_PREFIX);
             }
          //filter starts with a limitfile subset index
-         else if (*options >= '0' && *options < '9')
```
Comment (vijaysun-omr) [inline compiler/control/OMROptions.cpp]: Good catch :)

**Pull request merged**